tests/testthat/test-conditional-transform.R

```diff
@@ -0,0 +1,199 @@
+context("Conditional transform")
+
+with_mock_crunch({
+    ds <- loadDataset("test ds")
+    ds2 <- loadDataset("ECON.sav")
+
+    test_that("conditionalTransform input validation", {
+        expect_error(conditionalTransform("gender", data = ds),
+                     'no conditions have been supplied; please supply formulas as conditions.')
+        expect_error(conditionalTransform("bar"~"foo", data = ds),
+                     'The LHS provided is not a CrunchLogicalExpr: "bar"')
+        expect_error(conditionalTransform(gender~"foo", data = ds, type="unknown"),
+                     paste0("type must be either ", dQuote("categorical"), ", ", dQuote("text"), ", or ", dQuote("numeric")))
+        expect_error(conditionalTransform(gender~"foo", data = ds, type="text",
+                                          categories=c("foo", "bar")),
+                     paste0("type is not ", dQuote("categorical"), " ignoring ",
+                     dQuote("categories")))
+        # check that we can't reference two different datasets
+        expect_error(conditionalTransform(ds$gender == 'Male' ~ "foo",
+                                          ds2$gender == 'Male' ~ "foo",
+                                          name = "new"),
+                     paste0("There is more than one dataset referenced. Please ",
+                            "supply only one."))
+    })
+
+    test_that("conditionalTransform works with categories", {
+        expect_silent(new_var <- conditionalTransform(gender == "Male" ~ textVar, data = ds))
+        expect_equal(new_var$values, c(-1, -1, -1, -1, -1, -1, 2, -1, -1, -1,
+                                       3, -1, 4, -1, -1, -1, -1, -1, -1, -1, 1,
+                                       6, 3, -1, 5))
+        expect_equal(new_var$type, "categorical")
```

Comment:
This seems odd to me. Your expression says "i want a variable where the values are `textVar` where gender is Male", but your resulting variable is categorical and not text?